src/Service/MetaDataService/MetaDataServiceBootstrap.cs

```diff
@@ -0,0 +1,32 @@
+using Fido2NetLib;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Passwordless.Service.MetaDataService;
+
+public static class MetaDataServiceBootstrap
+{
+    public static void AddMetaDataService(this WebApplicationBuilder builder)
+    {
+        var configuration = builder.Configuration;
+
+        var httpClientBuilder = builder.Services.AddHttpClient(nameof(Fido2MetadataServiceRepository), client =>
+        {
+            client.BaseAddress = new Uri("https://mds3.fidoalliance.org/");
+        });
+
+        if (configuration["SelfHosted"] == "true" && configuration["fido2:mds:mode"] != "Online")
```

Comment:
For self-hosting, we may want to give the customer the option to only load from `mds.jwt` found in `/etc/bitwarden_passwordless`, which would be mapped as a persistent volume. This is what `OfflineCacheHandler` is for.

`CacheHandler` will turn cache a successful response in a local folder `.mds-cache`, so we can recover upon failure. The distributed cache will expire daily, upon expiry, it will refresh the cache automatically.

Initial attempt was to use a backgroundservice and call the `Fido2MetadataServiceRepository` to update the blob, but due to deserialization/serialization errors, caching it locally wouldn't work. I was then using the file repository with the DistributedCacheMetadataService to load the cached files.

Either way, both solutions are roughly working the same.